Product.cs

```diff
@@ -8,52 +8,54 @@ namespace Price_Calculator
 {
     public class Product
     {
+        
         public String name { get; set; }
+
         public int upc { get; set; }
 
         public double price { get; set; }
 
-        public double flat_rate_task { get; private set; }
+        public double total_price { get; set; }
 
-        public void SetTaskPercentage(double task_percentage)
-        {
-            flat_rate_task = (task_percentage / 100);
-        }
+        public double flat_rate_tax { get; private set; }
+        public double discount { get; set; }
 
         public Product()
         {
             name = "";
             upc = 0;
             price = 0.0;
-            flat_rate_task = 0.2;
+            flat_rate_tax = 0.2;
+            discount = 0;
+            ProductUtilities.CreateInstance(this);
+            this.total_price=this.CalculateTotalPrice();
+
         }
-        public Product(String name, int upc, double price, double flat_rate_task)
+        public Product(String name, int upc, double price, double flat_rate_tax, double discount)
```

Comment:
here you're passing the tax rate and discount to each product individually.
Suppose we had like 100 products, and we wanted to change the tax rate from 20 to another value? how'd you solve it?